### DISCUSSION TITLE
Will blazor-binding be supported in the future?

### DISCUSSION THREAD

**Discussion by CeSun** [unoplatform/uno#10170] [General]
Xamarin-blazor-binding: https://learn.microsoft.com/en-us/mobile-blazor-bindings/
BlazorBindings.Maui: https://github.com/Dreamescaper/BlazorBindings.Maui

Razor is more flexible than XAML，Hopefully this feature will be added in the future.

**Comment by jeromelaban** [comment]
This is a concept we already have experimented on, yet the razor syntax has a lot of missing features when applied to the way the WinUI object model works (e.g. attached properties).

What do you find more flexible in this model? Is it the fact that the code is present at the end of the file?

**Reply by CeSun** [reply]
Simpler list of controls instead of using template, more concise viewmodel, no need to manually trigger any events, or add attributes of source generators.

Like:
```razor
<ContentView>
  <Entry Text="Name" bind-text="@inputName"></Entry>
  <Button OnClick="@Insert"></Button >
  @foreach(var item in MyList)
  {
         <Lable Text="@item"></Lable>
  }

</ContentView>

@code
{
   string inputName;

   List<string> MyList = new ();

   void OnClick()
   {
      MyList.Add(inputName);
      inputName="";
   }
}
```

**Reply by jeromelaban** [reply]
I see. This kind of system is actually very expensive in a XAML world, because items are explicitly generated when the markup is reevaluated. It's also quite memory intensive as it regenerates objects when there's no need to.

The corresponding XAML is not that much different and is able to be more memory and CPU efficient, particularly with the reuse of items in the `ItemsControl`.

```xaml
<StackPanel>
    <TextBox Header="Name" Text="{x:Bind InputName, Mode=TwoWay}" />
    <Button Content="Insert" Click="{x:Bind OnInsert}" />

    <ItemsControl ItemsSource="{x:Bind MyList}">
        <!-- 
          For the simple case above, the template is not needed, but this 
          generally what is used because is a simple label is rarely used alone in a larger app
        -->
        <ItemsControl.ItemTemplate>
            <DataTemplate>
               <TextBlock Text="{Binding}" />
            </DataTemplate>
        </ItemsControl.ItemTemplate>
    </ItemsControl>
</StackPanel>
```

```csharp
public sealed partial class MainPage : Page
{
    public MainPage() => this.InitializeComponent();

    ObservableCollection<string> MyList = new();
    string InputName;

    void OnInsert()
    {
        MyList.Add(InputName);
        InputName = null;
        Bindings.Update(); // Reevaluate InputName without INotifyPropertyChanged.
    }
}
```

**Reply by CeSun** [reply]
My friends, there is another question unrelated to this topic, what should I do to switch the theme of uno platform? I don't understand the documentation, which package should I add to which project and which package should I remove?

**Reply by jeromelaban** [reply]
> My friends, there is another question unrelated to this topic, what should I do to switch the theme of uno platform? I don't understand the documentation, which package should I add to which project and which package should I remove?

It's best if you open another discussion for this, to avoid mixing up topics.
